client/api/filesystem/v1alpha1/api.proto

```diff
@@ -4,25 +4,156 @@ package v1alpha1;
 
 import "github.com/kubernetes-csi/csi-proxy/client/api/errors.proto";
 
-// FIXME: this is just an almost empty service, not actually implemented; it's just
-// here as an example of what API group and versions will look like. It will actually
-// be implemented in a later patch.
-
 service FileSystem {
-    // PathExists checks if the given path exists on the host.
+    // PathExists checks if the parameter path exists on the host filesystem.
     rpc PathExists(PathExistsRequest) returns (PathExistsResponse) {}
+
+    // Mkdir creates a directory at the requested path on the host filesystem.
+    rpc Mkdir(MkdirRequest) returns (MkdirResponse) {}
+
+    // Rmdir removes the directory at the requested path on the host filesystem.
+    // This may be used for unlinking a symlink created through LinkPath.
+    rpc Rmdir(RmdirRequest) returns (RmdirResponse) {}
```

Comment:
if the dir has content in it, it will delete the content too?

Reply:
I think we can add a parameter to differentiate the two options - remove if non-empty vs remove iff directory is empty.

Reply:
what is the expected behavior for non-empty dir?

Reply:
Should be an error.

Reply:
I like the idea of the parameter. Are you planning on implementing this in this PR? Or follow up?

Reply:
I will ammend in this PR.